src/EditorFeatures/CSharpTest/UseInterpolatedVerbatimString/UseInterpolatedVerbatimStringCodeFixTests.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.UseInterpolatedVerbatimString;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.UseInterpolatedVerbatimString
+{
+    [Trait(Traits.Feature, Traits.Features.CodeActionsUseInterpolatedVerbatimString)]
+    public class CSharpUseInterpolatedVerbatimStringCodeFixTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    {
+        internal override (DiagnosticAnalyzer, CodeFixProvider) CreateDiagnosticProviderAndFixer(Workspace workspace)
+            => (null, new CSharpUseInterpolatedVerbatimStringCodeFixProvider());
+
+        [Fact]
+        public async Task Simple()
+        {
+            await TestInRegularAndScript1Async(
+@"class C
+{
+    void M()
+    {
+        var s = @[||]$""hello"";
```

Comment:
can you have a test when my care is here:

```c#
var s = @$""hello""[||];
```

I want to make sure i can easily fix up this typo after i make it when i'm at the end of the string.  Thanks!

Reply:
I like the idea, but I'm wary of extending the diagnostic to cover the whole string (large area of red squiggles). 
Initially, I thought I might be able to add some additional locations to the diagnostic, but it looks like the logic for matching the selected span to the list of diagnostics is pretty straightforward and doesn't account for some other locations.
 Maybe I could generate a hidden diagnostic for the length of the string?

Reply:
> Maybe I could generate a hidden diagnostic for the length of the string?

Yes.  i think that could work!   But it's also probably not that important :)  

Here's why:  i think it's fine for the span to be just at the beginning.  The actual lightbulb infrastructure will still let you bring it up even if you're at the end of hte string.  it's just that our test harness doesn't really have a way to represent that.  So i don't think you need to do anything here.